python/sqlflow_submitter/xgboost/pai_rabit.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2020 The SQLFlow Authors. All rights reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import http.client
+import socket
+import threading
+import time
+from contextlib import closing
+from http.server import BaseHTTPRequestHandler, HTTPServer
+
+from sqlflow_submitter.xgboost.tracker import RabitTracker
+
+
+class PaiWorker():
+    @staticmethod
+    def read_tracker_port(host, port, ttl=900):
+        started = int(time.time())
+        each_sleep = 30
+        while True:
+            if int(time.time()) - started > ttl:
+                raise Exception(
+                    "failed to read the tracker port in {} seconds".format(
+                        ttl))
+            conn = None
+            try:
+                conn = http.client.HTTPConnection(host=host,
+                                                  port=port,
+                                                  timeout=60)
+                conn.request('GET', '/')
+                rsp = conn.getresponse()
+                return str(rsp.read())
+            except Exception:
+                time.sleep(each_sleep)
+            finally:
+                if conn is not None:
+                    conn.close()
+
+    @staticmethod
+    def gen_envs(host, port, ttl, nworkers, task_id):
+        tracker_port = PaiWorker.read_tracker_port(host, port, ttl)
+        if tracker_port == "":
+            raise Exception("bad response from tracker, empty port")
+        return [
+            str.encode('DMLC_NUM_WORKER=%d' % nworkers),
+            str.encode('DMLC_TRACKER_URI=%s' % host),
+            str.encode('DMLC_TRACKER_PORT=%s' % tracker_port),
+            str.encode('DMLC_TASK_ID=%d' % task_id)
+        ]
+
+
+class PaiTracker(object):
```

Comment:
This tracker is for XGBoost distributed training, so call it `PaiXGBoostTracker` and `PaiXGBoostWorker`?